builtin/providers/google/resource_storage_bucket.go

```diff
@@ -0,0 +1,154 @@
+package google
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/storage/v1"
+)
+
+func resourceStorageBucket() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStorageBucketCreate,
+		Read:   resourceStorageBucketRead,
+		Update: resourceStorageBucketUpdate,
+		Delete: resourceStorageBucketDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"predefined_acl": &schema.Schema{
+				Type:     schema.TypeString,
+				Default:  "private",
+				Optional: true,
+				ForceNew: true,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"force_destroy": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceStorageBucketCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Get the bucket and acl
+	bucket := d.Get("name").(string)
+	acl := d.Get("predefined_acl").(string)
+	location := d.Get("location").(string)
+
+	// Create a bucket, setting the acl, location and name.
+	sb := &storage.Bucket{Name: bucket, Location: location}
+	res, err := config.clientStorage.Buckets.Insert(config.Project, sb).PredefinedAcl(acl).Do()
+
+	if err != nil {
+		fmt.Printf("Error creating bucket %s: %v", bucket, err)
+		return err
+	}
+
+	log.Printf("[DEBUG] Created bucket %v at location %v\n\n", res.Name, res.SelfLink)
+
+	// Assign the bucket ID as the resource ID
+	d.SetId(res.Id)
+
+	// Add extra metadata about the bucket.
+	d.Set("name", res.Name)
+	d.Set("predefined_acl", res.Acl)
```

Comment:
This works and what it does is to set those attributes for that resource. They are easily visible in the tfstate file. I am not sure what everything these are used for, but I would say terraform at least uses that to monitor changes in state. I am storing name, acl, location - as these are fields configured by user - to whatever google returns in the response.  The result is (this is a snipplet from tfstate):
```
"google_storage_bucket.michael-testbucket": {
                    "type": "google_storage_bucket",
                    "primary": {
                        "id": "michaeltestbucket2",
                        "attributes": {
                            "force_destroy": "true",
                            "id": "michaeltestbucket2",
                            "location": "US",
                            "name": "michaeltestbucket2",
                            "predefined_acl": "projectprivate"
                        }
                    }
                }
```
Why use both `name` and `Id` ? Because maybe in the future GCS can start returning st. else than name as ID...
Would be great to also get some terraform feedback on proper behaviour here. The AWS S3 provider sets only ID on creation and then sets all other attributes in bucketRead function. Terraform seems to do bucketRead (or, more abstractly, resource read) every time before it is about to modify the resource. I presume this is to find out the real state of that resource to determine if it's ready for update.